### PR TITLE
Provide example alerts in Cookie scan rules

### PR DESCRIPTION
**Pull request by martinspielmann**
The CWE id was already managed for these Alerts/ScanRules and within ZAP they were already presented to the user.
However, when visiting the alert detail pages online (https://www.zaproxy.org/docs/alerts/), CWE (and WASC) links were not generated correctly.

The reason is, that the id values were set inline to the alert.
For the script which generates the detail pages however, the getter methods for the ids have to be added publicly to the class.

Note: if this solves the issue, I'll vote for another follow up issue to add this change to the PluginPassiveScanner parent class,
so that the docs will be generated consistently over all Passive Scan Rules.

Fixes zaproxy/zaproxy#6140.

### CODE REVIEW

**Comment by thc202** [comment]
> Note: if this solves the issue, I'll vote for another follow up issue to add this change to the PluginPassiveScanner parent class, so that the docs will be generated consistently over all Passive Scan Rules.

It does, but there's a new way to provide the alert details, more details in zaproxy/zaproxy#6119 (this could be changed to do that if you want).

**Comment by kingthorin** [comment]
To address the DCO requirement you'll need to sign-off the commit(s):

- https://github.com/zaproxy/zaproxy/blob/develop/CONTRIBUTING.md#developer-certificate-of-origin
- https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff

**Comment by kingthorin** [comment]
Also needs a spotlessApply: `./gradlew :addOns:pscanrules:spotlessApply`

File: addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/CookieHttpOnlyScanRule.java [hunk 1]
```diff
@@ -134,4 +134,28 @@ private Model getModel() {
     protected void setModel(Model model) {
         this.model = model;
     }
+
+    /**
+     * Gets the CWE ID of the issue(s) raised by the scanner.
+     *
+     * @return the CWE ID,
+     * @see <a href="https://cwe.mitre.org/index.html">CWE - Common Weakness Enumeration</a>
+     */
```
Comment (thc202) [inline addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/CookieHttpOnlyScanRule.java:143]: Not worth adding the JavaDoc.

**Comment by martinspielmann** [comment]
Thanks @kingthorin and @thc202 for the hints, sorry saw that  too late :monkey: Will update PR asap

**Comment by kingthorin** [comment]
No worries.

**Comment by martinspielmann** [comment]
Regarding the comment:
> It does, but there's a new way to provide the alert data, more details in zaproxy/zaproxy#6119 (this could be changed to do that if you want).

After reading through, I think it would indeed be cleaner to go for the new `List<Alert> getExampleAlerts()` method. Should I add that here already or would it be preferable to have that in another PR clearly referencing to the introduction of getExampleAlerts

**Comment by kingthorin** [comment]
I'm fine with it being here. You can always squash this.

**Comment by martinspielmann** [comment]
> I'm fine with it being here. You can always squash this.

Ok ready :)

File: addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/CookieHttpOnlyScanRule.java [hunk 2]
```diff
@@ -134,4 +135,30 @@ private Model getModel() {
...
+
+    private int getWascId() {
+        // WASC Id 13 - Info leakage)
+        return 13;
+    }
+
+    public List<Alert> getExampleAlerts() {
+        List<Alert> alerts = new ArrayList<Alert>();
+        Alert alert =
+                newAlert()
```
Comment (thc202) [inline addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/CookieHttpOnlyScanRule.java:152]: This should call existing method, avoids maintaining the same info in two places (which is error prone). There's another PR with similar changes: https://github.com/zaproxy/zap-extensions/pull/2517/files#diff-2587ce4776bf99dbb14cf9284389ba13
The methods `getCweId()` and `getWascId()` would no longer be needed.

**Review by kingthorin — COMMENTED** [review]
One minor thing, and it’s an existing issue

**Review by kingthorin — APPROVED** [review]
Thanks

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**